protocol/common/message_segmentation.go

```diff
@@ -0,0 +1,361 @@
+package common
+
+import (
+	"bytes"
+	"math"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jinzhu/copier"
+	"github.com/klauspost/reedsolomon"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/protobuf"
+	v1protocol "github.com/status-im/status-go/protocol/v1"
+)
+
+var ErrMessageSegmentsIncomplete = errors.New("message segments incomplete")
+var ErrMessageSegmentsAlreadyCompleted = errors.New("message segments already completed")
+var ErrMessageSegmentsInvalidCount = errors.New("invalid segments count")
+var ErrMessageSegmentsHashMismatch = errors.New("hash of entire payload does not match")
+var ErrMessageSegmentsInvalidParity = errors.New("invalid parity segments")
+
+const (
+	segmentsParityRate = 0.125
+	segmentsMaxCount   = 256
+)
+
+type SegmentMessage struct {
+	*protobuf.SegmentMessage
+}
+
+func (s *SegmentMessage) IsValid() bool {
+	return s.SegmentsCount >= 2 || s.ParitySegmentsCount > 0
```

Comment:
Should the 2 be in a constant? Is this something that will be configurable or something?

Reply:
We don't create segments if the message fits within the limit; the segment count must be at least two. The condition will never change. Creating a constant for that may be overkill.